Library/Homebrew/api/cask.rb

```diff
@@ -79,21 +79,25 @@ def self.cached_json_file_path
           .returns([T.any(T::Array[T.untyped], T::Hash[String, T.untyped]), T::Boolean])
       }
       def self.fetch_api!(download_queue: nil, stale_seconds: Homebrew::EnvConfig.api_auto_update_secs.to_i)
-        Homebrew::API.fetch_json_api_file DEFAULT_API_FILENAME, stale_seconds:, download_queue:
+        skip_download = cache.fetch("downloaded", false)
```

Comment:
Could this check for the existence of the file on disk instead?

There's already e.g. mtime handling in `API.fetch_json_api_file` that must not be working as expected here.

Reply:
See https://github.com/Homebrew/brew/pull/20619#issuecomment-3243209696 for a better explanation